packages/client-github/src/index.ts

```diff
@@ -0,0 +1,236 @@
+    private repoPath: string;
+
+    constructor(runtime: AgentRuntime) {
+        this.runtime = runtime;
+        this.config = {
+            owner: runtime.getSetting("GITHUB_OWNER") as string,
+            repo: runtime.getSetting("GITHUB_REPO") as string,
+            branch: runtime.getSetting("GITHUB_BRANCH") as string,
+            path: runtime.getSetting("GITHUB_PATH") as string,
+            token: runtime.getSetting("GITHUB_API_TOKEN") as string,
+        };
+        this.octokit = new Octokit({ auth: this.config.token });
+        this.git = simpleGit();
+        this.repoPath = path.join(
+            process.cwd(),
+            ".repos",
+            this.config.owner,
+            this.config.repo
+        );
+    }
+
+    async initialize() {
+        // Create repos directory if it doesn't exist
+        await fs.mkdir(path.join(process.cwd(), ".repos", this.config.owner), {
+            recursive: true,
+        });
+
+        // Clone or pull repository
+        if (!existsSync(this.repoPath)) {
+            await this.git.clone(
+                `https://github.com/${this.config.owner}/${this.config.repo}.git`,
+                this.repoPath
+            );
+        } else {
+            const git = simpleGit(this.repoPath);
+            await git.pull();
+        }
+
+        // Checkout specified branch if provided
+        if (this.config.branch) {
+            const git = simpleGit(this.repoPath);
+            await git.checkout(this.config.branch);
+        }
+    }
+
+    async createMemoriesFromFiles() {
+        console.log("Create memories");
+        const searchPath = this.config.path
+            ? path.join(this.repoPath, this.config.path, "**/*")
+            : path.join(this.repoPath, "**/*");
+
+        const files = await glob(searchPath, { nodir: true });
+
+        for (const file of files) {
+            const relativePath = path.relative(this.repoPath, file);
+            const content = await fs.readFile(file, "utf-8");
+            const contentHash = createHash("sha256")
+                .update(content)
+                .digest("hex");
+            const knowledgeId = stringToUuid(
+                `github-${this.config.owner}-${this.config.repo}-${relativePath}`
+            );
+
+            const existingDocument =
+                await this.runtime.documentsManager.getMemoryById(knowledgeId);
+
+            if (
+                existingDocument &&
+                existingDocument.content["hash"] == contentHash
+            ) {
+                continue;
+            }
+
+            console.log(
+                "Processing knowledge for ",
+                this.runtime.character.name,
+                " - ",
+                relativePath
+            );
+
+            const memory: Memory = {
+                id: knowledgeId,
+                agentId: this.runtime.agentId,
+                userId: this.runtime.agentId,
+                roomId: this.runtime.agentId,
```

Comment:
should we make this based on a repo?

Reply:
right now this is the same config that is used for global knowledge base defined in the character config